From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Tue, 17 Feb 2015 10:45:30 -0500
Subject: tool_lib.c: ensure output string is always set

Ensure that val2str always writes to its output buffer,
which is static and shared between all PVs.
Should make bugs like lp:1418526 more obvious.
---
 src/catools/tool_lib.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/catools/tool_lib.c b/src/catools/tool_lib.c
index 6ad1a8b..ce41fde 100644
--- a/src/catools/tool_lib.c
+++ b/src/catools/tool_lib.c
@@ -119,6 +119,7 @@ char *val2str (const void *v, unsigned type, int index)
         strcpy (str, "*** invalid type");
         return str;
     }
+    strcpy (str, "!!!");
 
     base_type = type % (LAST_TYPE+1);
 
